simulation-tests/src/test/java/org/apache/druid/testing/simulate/embedded/EmbeddedBroker.java

```diff
@@ -0,0 +1,76 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.druid.testing.simulate.embedded;
+
+import com.google.inject.Injector;
+import com.google.inject.Module;
+import org.apache.druid.cli.CliBroker;
+import org.apache.druid.cli.ServerRunnable;
+import org.apache.druid.java.util.common.lifecycle.Lifecycle;
+import org.apache.druid.query.DruidProcessingConfigTest;
+import org.apache.druid.utils.RuntimeInfo;
+
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ * Embedded mode of {@link CliBroker} used in simulation tests.
+ */
+public class EmbeddedBroker extends EmbeddedDruidServer
```

Comment:
this is the framework part: I think these classes should be under `main` of this module